src/devices/bus/qbus/qbus.cpp

```diff
@@ -124,6 +124,13 @@ void qbus_device::device_start()
 
 void qbus_device::device_reset()
 {
+        device_qbus_card_interface *entry = m_device_list.first();
+
+        while (entry)
+        {
+                entry->device_reset();
+                entry = entry->next();
+        }
```

Comment:
This is going to cause problems.  You generally shouldn’t be calling `device_reset` directly as it doesn’t call `interface_pre_reset`/`interface_post_reset` for mixins, and it doesn’t reset child devices automatically.